src/tree_edit.h

```diff
@@ -231,6 +231,28 @@ void *ly_realloc(void *ptr, size_t size);
 #define LY_ARRAY_FREE(ARRAY) \
         if (ARRAY){free((LY_ARRAY_COUNT_TYPE*)(ARRAY) - 1);}
 
+/**
+ * @brief Remove item from array based on value
+ *
+ * @param[in, out] ARRAY A ([sized array](@ref sizedarrays)) to be modified.
+ * @param[in] INDEX The item position to be removed.
+ */
+#define LY_ARRAY_REMOVE_VALUE(ARRAY, VALUE) \
```

Comment:
Seems it keeps the original order, is that necessary, why not just replace the removed value with the last value? If you think it is, wouldn't it be better to call `memmove()` once instead?

Reply:
Done